addons/project/tests/test_project_sharing_ui.py

```diff
@@ -66,3 +66,53 @@ def test_02_project_sharing(self):
             })],
         })
         self.start_tour("/my/projects", 'portal_project_sharing_tour', login='georges1')
+
+    def test_blocked_task_with_project_sharing_string_portal(self):
+        """
+        Ensure the portal user shows the message 'This task is currently blocked...'.
+        Flow:
+            - Activated Task Dependencies in a portal project
+            - Create a 'New' task stage
+            - Create a project(Test Project)
+            - Create task(Test Task)
+            - Create a task with a Blocked task (Test Task)
+            - Ensure the portal user receives the message 'This task is currently blocked..'.
+        """
+
+        self.project_portal.write({
+            'allow_task_dependencies': True,
+            'collaborator_ids': [
+                Command.create({'partner_id': self.partner_portal.id}),
+            ],
+        })
+
+        task_stage = self.env['project.task.type'].create({
+            'name': 'NEW',
+        })
```

Comment:
do we need to create a new stage for the ``project_portal`` project ? :thinking: from the setup class it already has 2 I believe :thinking: "To-Do" and "Done"

Reply:
Hello @pkyriakou 
In the setup, stages were indeed created but that is for the project, including "To-Do" and "Done". 
However, in our specific context, we require stages for individual tasks within the project, becacuse
without a stage for tasks, the interface appear shrink, and clicking on individual tasks not be possible :slightly_smiling_face: .

Reply:
the type_ids field defined in project.project is of ``project.task.type`` model thus its for the tasks in within a project :thinking: 
```py
type_ids = fields.Many2many('project.task.type', 'project_task_type_rel', 'project_id', 'type_id', string='Tasks Stages')
```
Am I understanding something wrong here :thinking:

Reply:
Hello @pkyriakou 
indeed its my bad, Changes done :+1: